includes/Level.hpp

```diff
@@ -100,18 +100,37 @@ class Level
      */
     bool allowPlayerMovement(const short& movement) const;
 
-private:
+    /**
+     * @brief get the current player cell type; this getter is used into the
+     * game controller to execute the correct current player cell event action
+     *
+     * @return const char&
+     */
+    const char& getPlayerCellType();
 
     /**
      * @brief check if the expected cell after the movement is a wall cell,
-     * if yes, show the wall cell and forbid the movement
+     * if yes, show the wall cell and forbid the movement; this function is
+     * called by the game controller when the player is allowed to move, we
+     * check if the user is currently in collision with a wall; if there is
+     * a collision, the wall is shown
      *
      * @param movement the movement number
      *
      * @return bool
      */
     bool detectWalls(const short& movement) const;
 
+    /**
+     * @brief update the current player cell to an empty cell, whatever the
+     * previous type; this method is called by the game controller to empty
+     * a cell when the player leaves it; this method automatically force the 
+     * load of a new texture for the player cell
+     */
+    void emptyPlayerCell();
```

Comment:
a lot of method of this class are called by the game controller, make the game controller as a "friend" of this class and make them as private